### DISCUSSION TITLE
[QUESTION] Should I be able to use this playbook for single-node server installation? - AWS Install

### DISCUSSION THREAD

**Discussion by neilknowscomputers** [techno-tim/k3s-ansible#473] [Questions] [answered]
## Expected Behavior

I should be able to run k3s on a single machine acting as the `master ` and `node` as I can do with a single-node server installation using the k3s binary

## Current Behavior

When I run the playbook, the agent node fails to connect to the master on `TASK [k3s_server : Verify that all nodes actually joined (check k3s-init.service if this fails`

## Steps to Reproduce

1. Set the same IP/host for `[master]` and `[node]` in `hosts.ini`
2. Run `ansible-playbook site.yml -i inventory/my-cluster/hosts.ini`

## Context (variables)

Operating system: Ubuntu Jammy 22.04

Hardware: AMD64 - AWS EC2 - 4 procs - 16 GB mem - 100 GB HD

### Variables Used

No change to default configs

### Hosts

`host.ini`

```ini
[master]
my-aws-ec2-host

[node]
my-aws-ec2-host

[k3s_cluster:children]
master
node
```

## Possible Solution

- [x] I've checked the [General Troubleshooting Guide](https://github.com/techno-tim/k3s-ansible/discussions/20)

**Comment by timothystewart6** [comment, marked as answer]
Yes, you should, and we [actually test this in CI](https://github.com/techno-tim/k3s-ansible/actions/runs/8224392662/job/22488207713).  This might be related to your cloud host

**Reply by neilknowscomputers** [reply]
Great! Thanks! Issue is probably on my end either in config or VM set up. I'll troubleshoot further.

I'm looking at the details of the CI test as a reference and comparing to my setup. Looking at the [molecule.yml](https://github.com/techno-tim/k3s-ansible/blob/master/molecule/single_node/molecule.yml) file for the single node scenario its not clear to me what inventory is being used. Can you point me in the right direction?
